### PR TITLE
Add create_sobject to support creation of Salesforce object

### PR DESCRIPTION
**Pull request by epinault**
Added per #4 support to create object in Salesforce

### CODE REVIEW

File: lib/ex_force.ex [hunk 1]
```diff
@@ -169,6 +169,20 @@ defmodule ExForce do
...
   @doc """
+  Creates a SObject.
+
+  See [SObject Rows](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_sobject_retrieve.htm)
+  """
+  @spec create_sobject(sobject_name, map, config_or_func) :: :ok | {:error, any}
+  def create_sobject(name, attrs, config \\ default_config()) do
+    case request_post("/sobjects/#{name}/", Poison.encode!(attrs), config) do
+      {201, raw} ->
+        {:ok, extract_create_id(raw)}
```
Comment (chulkilee) [inline lib/ex_force.ex]: Could you run `mix format`? I believe the formatter will make single line vs multil line for pattern match in `case` consistent.

See https://elixirforum.com/t/running-the-new-elixir-formatter/9344/36 to install master elixir and run it, without interrupting your elixir version.

File: lib/ex_force.ex [hunk 2]
```diff
@@ -236,6 +250,10 @@ defmodule ExForce do
   def query_all_stream(soql, config \\ default_config()),
     do: start_query_stream(&query_all/2, soql, config)
 
+  defp extract_create_id(resp = %{"id" => id, "success" => "true"}) do
```
Comment (chulkilee) [inline lib/ex_force.ex]: Let's make it inline in caller side, like this:

```elixir
{201, %{"id" => id, "success" => true}} -> {:ok, id}
```

Reasons & Note:

- The extraction implementation is too simple to be in a separate function - just plain pattern match would work
- Do all create requests return responses in this structure? Is this structure only used by create? I'd defer making abstraction..

File: lib/ex_force.ex [hunk 3]
```diff
@@ -169,6 +169,20 @@ defmodule ExForce do
   end
 
   @doc """
+  Creates a SObject.
+
+  See [SObject Rows](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_sobject_retrieve.htm)
```
Comment (chulkilee) [inline lib/ex_force.ex]: The actual doc is at https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_sobject_basic_info.htm

File: test/ex_force_test.exs [hunk 4]
```diff
@@ -286,6 +286,64 @@ defmodule ExForceTest do
...
+  test "create_sobject - success", %{bypass: bypass} do
+    Bypass.expect_once(bypass, "POST", "/services/data/v40.0/sobjects/Account/", fn conn ->
+      {:ok, raw, conn} = Conn.read_body(conn)
+      expected_body = %{"FirstName" => "first_name"}
+      ^expected_body = Poison.decode!(raw)
+      resp_body = """
+      {
+        "id": "001D000000IqhSLIAZ",
+        "errors": [],
+        "success": "true"
```
Comment (chulkilee) [inline test/ex_force_test.exs]: When I tested it manually for v40.0, it returns `warnings` as well. Could you check that out? I'd like to keep it close to real response as much as possible.

File: lib/ex_force.ex (hunk 3 above)
Comment (epinault) [reply lib/ex_force.ex]: ah yea , forgot to updated it.

File: lib/ex_force.ex (hunk 1 above)
Comment (epinault) [reply lib/ex_force.ex]: ah, I have 1.5.2. I ll install a master release of elixir then  (I am using asdf for Elixir verison management so I ll figure it out). I heard for the formatter but did not try it yet :)

File: lib/ex_force.ex (hunk 2 above)
Comment (epinault) [reply lib/ex_force.ex]: Excuse my newbie experience in Elixir :) been only writting it for a month

As far as I can tell only the create seems to have that structure. And even more bizzarre, the success is never sent on error. nor the errors either. they come in a different format. So I am not seeing an documentation on that either beside that one page

File: test/ex_force_test.exs (hunk 4 above)
Comment (epinault) [reply test/ex_force_test.exs]: hum. I ll try it again but did not see any warnings. let me check  But that is what they document at https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_sobject_create.htm

Comment (epinault) [reply test/ex_force_test.exs]: you are right, I see the warnings field too. I ll fix  that but poor documentation on salesforce end. Cannot see it in their Rest doc...

**Comment by epinault** [comment]
done!

**Comment by chulkilee** [comment]
🎉 Thanks a lot!

**Pull request merged**